packages/api/src/globalContext.ts

```diff
@@ -1,13 +1,63 @@
+// AWS Lambda run each request in a new process,
+// a process is not reused until a request is completed.
+//
+// Which means that each `global.context` is scoped to the lifetime of each request.
+
+// However when not in AWS Lambda, NodeJS is single-threaded, you must use the
+// per-request global context, otherwise you risk a race-condition
+// where one request overwrites another's global context.
+//
+// Alternatively only use the local `context` in a graphql resolver.
+
 export interface GlobalContext {
-  [key: string]: any
+  [key: string]: unknown
+}
+
+import { AsyncLocalStorage } from 'async_hooks'
+
+export let asyncLocalStorage:
+  | AsyncLocalStorage<Map<string, GlobalContext>>
+  | undefined = undefined
+export const initGlobalContext = () => {
+  if (process.env.SAFE_GLOBAL_CONTEXT !== '1') {
```

Comment:
Where is this ENV var set?

Reply:
I'm making the per request context the default.